### ISSUE TITLE
Error index

### ISSUE THREAD

**Issue by kdy1** [swc-project/swc#4143]
### Describe the feature

We can have something like https://doc.rust-lang.org/error-index.html and describe various errors.
This is not a high priority, but it will be useful for people who do not have much experiences

**Comment by kwonoj** [comment]
I'm ++++1 for this. React have similar as well, difference is it doesn't emit any error message for the prod build (https://reactjs.org/blog/2016/07/11/introducing-reacts-error-code-system.html)

**Comment by kdy1** [comment]
We can use diagnostic id to create url, but we need a policy for error messages 🤣

**Comment by kwonoj** [comment]
Yes, this is major bikeshedding to go through all of existing errors + reserve futures with proper context mapping. It'll be definitely helpful still.
